### ISSUE TITLE
In app crash on v2.1.0

### ISSUE THREAD

**Issue by hanghang** [hardcore-sushi/DroidFS#236]
Steps to reproduce on my phone:
1. open a file (I tested pdf and jpg)
2. go back go file list
3. open another file, volume closed like a crash
it's  OK on v2.0.2

but on v2.1.0, logcat shows :
```
09-15 15:57:32.824 29574 29574 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{sushi.hardcore.droidfs/sushi.hardcore.droidfs.file_viewers.PdfViewer}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3825)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3971)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2389)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:210)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:299)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8261)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at java.lang.Class.isAssignableFrom(Class.java:589)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Parcel.readParcelableCreatorInternal(Parcel.java:4853)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Parcel.readParcelableInternal(Parcel.java:4766)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Parcel.readValue(Parcel.java:4532)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Parcel.readValue(Parcel.java:4312)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Parcel.-$$Nest$mreadValue(Unknown Source:0)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Parcel$LazyValue.apply(Parcel.java:4410)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Parcel$LazyValue.apply(Parcel.java:4369)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.BaseBundle.getValueAt(BaseBundle.java:394)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.BaseBundle.getValue(BaseBundle.java:374)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.BaseBundle.getValue(BaseBundle.java:357)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.BaseBundle.get(BaseBundle.java:696)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.os.Bundle.getParcelable(Bundle.java:947)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.content.Intent.getParcelableExtra(Intent.java:8927)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at androidx.activity.ComponentDialog$$ExternalSyntheticApiModelOutline0.m(SourceFile:9)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at sushi.hardcore.droidfs.file_viewers.FileViewerActivity.onCreate(SourceFile:44)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8516)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8480)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1418)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3806)
09-15 15:57:32.824 29574 29574 E AndroidRuntime: 	... 12 more
```

hope the log helps.

**Comment by hardcore-sushi** [comment]
I can't reproduce. What's your Android version? Does it happen with both gocryptfs and cryfs volumes? Could you try to reproduce with a volume containing a single file and share it? Thanks

**Comment by hanghang** [comment]
I tested again, both gocryptfs and cryfs have the same problem.
My Android version is 13, customized by MIUI, they modified many standard APIs,
I will try on another phone tomorrow.

**Comment by hanghang** [comment]
I tried some other phones with standard Android OS, V2.1.0 confirmed OK.
Seems MIUI's customization caused the problem.
sorry for the confusion

**Comment by hardcore-sushi** [comment]
That's strange though. Do you have similar problems with other applications?

**Comment by hanghang** [comment]
No, other application run well, DroidFS v2.0.2 also works well.
Did you make use of some new APIs in the v2.1.x versions, especially in sushi.hardcore.droidfs.file_viewers package?
I guess the new APIs implements in MIUI have problems.

**Comment by hanghang** [comment]
I disabled  all these features and  reproduced on my phone again.
Notified  `androidx.activity.ComponentDialog...` exception in stackinfo, Maybe my ROM not supported androidx very well yet.


```
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at androidx.activity.ComponentDialog$$ExternalSyntheticApiModelOutline0.m(SourceFile:9)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at sushi.hardcore.droidfs.file_viewers.FileViewerActivity.onCreate(SourceFile:44)
```

full log
```
09-20 08:27:29.019 16646 16646 D MiuiMultiWindowAdapter: MiuiMultiWindowAdapter::getFreeformVideoWhiteListInSystem::LIST_ABOUT_SUPPORT_LANDSCAPE_VIDEO = [com.hunantv.imgo.activity, com.tencent.qqlive, com.qiyi.video, com.hunantv.imgo.activity.inter, com.tencent.qqlivei18n, com.iqiyi.i18n, tv.danmaku.bili]
09-20 08:27:29.022 16646 16682 D ViewContentFactory: initViewContentFetcherClass
09-20 08:27:29.022 16646 16682 I ContentCatcher: ViewContentFetcher : ViewContentFetcher
09-20 08:27:29.022 16646 16682 D ViewContentFactory: createInterceptor took 0ms
09-20 08:27:29.022 16646 16646 D AndroidRuntime: Shutting down VM
--------- beginning of crash
09-20 08:27:29.022 16646 16646 E AndroidRuntime: FATAL EXCEPTION: main
09-20 08:27:29.022 16646 16646 E AndroidRuntime: Process: sushi.hardcore.droidfs, PID: 16646
09-20 08:27:29.022 16646 16646 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{sushi.hardcore.droidfs/sushi.hardcore.droidfs.file_viewers.TextEditor}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3825)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3971)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2389)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:210)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:299)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8261)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at java.lang.Class.isAssignableFrom(Class.java:589)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Parcel.readParcelableCreatorInternal(Parcel.java:4853)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Parcel.readParcelableInternal(Parcel.java:4766)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Parcel.readValue(Parcel.java:4532)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Parcel.readValue(Parcel.java:4312)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Parcel.-$$Nest$mreadValue(Unknown Source:0)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Parcel$LazyValue.apply(Parcel.java:4410)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Parcel$LazyValue.apply(Parcel.java:4369)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.BaseBundle.getValueAt(BaseBundle.java:394)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.BaseBundle.getValue(BaseBundle.java:374)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.BaseBundle.getValue(BaseBundle.java:357)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.BaseBundle.get(BaseBundle.java:696)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.os.Bundle.getParcelable(Bundle.java:947)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.content.Intent.getParcelableExtra(Intent.java:8927)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at androidx.activity.ComponentDialog$$ExternalSyntheticApiModelOutline0.m(SourceFile:9)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at sushi.hardcore.droidfs.file_viewers.FileViewerActivity.onCreate(SourceFile:44)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8516)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8480)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1418)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3806)
09-20 08:27:29.022 16646 16646 E AndroidRuntime: 	... 12 more
09-20 08:27:29.019 16646 16646 W ardcore.droidfs: type=1400 audit(0.0:8980030): avc: denied { search } for name="miuilog" dev="dm-52" ino=960248 scontext=u:r:untrusted_app:s0:c190,c257,c512,c768 tcontext=u:object_r:data_log_file:s0 tclass=dir permissive=0
09-20 08:27:29.019 16646 16646 W ardcore.droidfs: type=1400 audit(0.0:8980031): avc: denied { search } for name="miuilog" dev="dm-52" ino=960248 scontext=u:r:untrusted_app:s0:c190,c257,c512,c768 tcontext=u:object_r:data_log_file:s0 tclass=dir permissive=0
09-20 08:27:29.019 16646 16646 W ardcore.droidfs: type=1400 audit(0.0:8980032): avc: denied { search } for name="miuilog" dev="dm-52" ino=960248 scontext=u:r:untrusted_app:s0:c190,c257,c512,c768 tcontext=u:object_r:data_log_file:s0 tclass=dir permissive=0
09-20 08:27:29.019 16646 16646 W ardcore.droidfs: type=1400 audit(0.0:8980033): avc: denied { search } for name="miuilog" dev="dm-52" ino=960248 scontext=u:r:untrusted_app:s0:c190,c257,c512,c768 tcontext=u:object_r:data_log_file:s0 tclass=dir permissive=0
09-20 08:27:29.019 16646 16646 W ardcore.droidfs: type=1400 audit(0.0:8980034): avc: denied { search } for name="miuilog" dev="dm-52" ino=960248 scontext=u:r:untrusted_app:s0:c190,c257,c512,c768 tcontext=u:object_r:data_log_file:s0 tclass=dir permissive=0
--------- beginning of system
09-20 08:27:29.023 16646 16646 W OOMEventManagerFK: Failed to mkdir /data/miuilog/stability/memleak/heapdump/
09-20 08:27:29.030 16646 16646 I Process : Process is going to kill itself!
09-20 08:27:29.030 16646 16646 I Process : java.lang.Exception
09-20 08:27:29.030 16646 16646 I Process : 	at android.os.Process.killProcess(Process.java:1344)
09-20 08:27:29.030 16646 16646 I Process : 	at com.android.internal.os.RuntimeInit$KillApplicationHandler.uncaughtException(RuntimeInit.java:181)
09-20 08:27:29.030 16646 16646 I Process : 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1073)
09-20 08:27:29.030 16646 16646 I Process : 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
09-20 08:27:29.030 16646 16646 I Process : 	at java.lang.Thread.dispatchUncaughtException(Thread.java:2306)
09-20 08:27:29.030 16646 16646 I Process : Sending signal. PID: 16646 SIG: 9
```

**Comment by hardcore-sushi** [comment]
Does it also happen when opening the internal camera or changing a volume's password?

**Comment by hardcore-sushi** [comment]
This version may fix the issue. Could you please tell me if it works fine when opening files, internal camera and when trying to change password?
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

SHA256:
f7aa592760ec4f30b1aa5baf8399028214b12a14592e7262b69239d6e8bda6b4 DroidFS-test.apk

Download link: https://drop.underworld.fr/r/mX6YVjd6Za#lwF2weI0LNvsiSQf7SnuoIXO92SvQqKlqqL2by5/9ZE=
-----BEGIN PGP SIGNATURE-----

iHUEARYIAB0WIQS2Tv6GzuHQVPCCFxGv44Q0SkXhOgUCZQ8UugAKCRCv44Q0SkXh
OrzTAQDurjQJn68F/I5Q1E6ytNQONhQBKvCc/ZAxlWCthAL1NAEAuU9CeA/0F8gH
Pd5fCcqmVF/TzfifkXB0LyGeBT8bzgw=
=B9Vz
-----END PGP SIGNATURE-----
```

**Comment by hanghang** [comment]
Wow, everything works great with this version.

**Comment by hardcore-sushi** [comment]
Great. The fix has been pushed: b503f134d5c4ecc97f3421530671d1e7cf1766fe. The next release should solve the issue.
